rxjs-1/speed.js

```diff
@@ -0,0 +1,28 @@
+const Rx = require('rxjs');
+
+const { getRace } = require('./race');
+const race = getRace();
+
+const carName = `Lightning McQueen`;
+
+const getCarSpeed = (race, car) => {
+    return new Rx.Observable(observer => {
+        let lastTime = 0;
+        let lastLocation = 0;
+        race.on('data', ({ time, carName, xLocation }) => {
+            if (car === carName && time - lastTime > 200) {
```

Comment:
As far as I read it, this implementation is computing the car speed _every 200ms_ with the last 200ms data (sequential approach) and not the car speed _at any moment_ with the last 200ms data (window approach), right?

How would you implement the _window approach_ ?

Reply:
Oh, I did not know you meant that by _time window_
To do this we need to store multiple values of time and location in an array, and each time we want to compute the speed, we find the value that is just above 200ms old in the array. (we can clean the array to remove useless old values)